pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+                 for inst in instance_store.iter_values()
+                 if inst.path.classname in clns_dict]
+
+        return insts
+
+    ####################################################################
+    #
+    #   EnumerateInstanceNames server request
+    #
+    ####################################################################
+
+    def EnumerateInstanceNames(self, namespace, ClassName):
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.EnumerateInstanceNames`
+
+        Get instance names for instances that match the path defined
+        by `ClassName` and return a list of the names.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          ClassName (:term:`string`):
+            Name of the class to be enumerated (case independent).
+
+        Returns:
+          A list of :class:`~pywbem.CIMInstanceName` objects that are the
+          enumerated instance paths, with its attributes set
+          as follows:
+
+          * `classname`: Name of the creation class of the instance.
+          * `keybindings`: Keybindings of the instance.
+          * `namespace`: Name of the CIM namespace containing the instance.
+          * `host`: `None`, indicating the WBEM server is unspecified.
+
+        Raises:
+            CIMError: CIM_ERR_CLASS_NOT_FOUND
+        """
+
+        assert isinstance(ClassName, six.string_types)
+
+        # Validate namespace and get instance CIM repository for this namespace
+        instance_store = self.get_instance_store(namespace)
+        class_store = self.get_class_store(namespace)
+
+        clns = self._get_subclass_list_for_enums(ClassName, namespace,
+                                                 class_store)
+
+        inst_paths = [inst.path for inst in instance_store.iter_values()
+                      if inst.path.classname in clns]
+
+        return_paths = [path.copy() for path in inst_paths]
+
+        return return_paths
+
+    ####################################################################
+    #
+    #   ExecQuery server request
+    #
+    ####################################################################
+
+    def ExecQuery(self, namespace, QueryLanguage, Query):
+        """
+        Implements a WBEM server responder for
+            :meth:`~pywbem.WBEMConnection.ExecQuery`
+
+        Executes the the query defined by the QueryLanguage and Query parameter
+        in the namespace defined.
+
+        NOTE: This operation currently returns CIM_ERR_NOT_SUPPORTED
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use query verbiage.

Reply:
DONE